src/standalone/Vegetation/Canopy.jl

```diff
@@ -59,6 +61,82 @@ include("./spatially_varying_parameters.jl")
 ########################################################
 # Convenience constructors for Canopy model components
 ########################################################
+## Soil Moisture Stress
+"""
+    PiecewiseMoistureStressModel{FT}(
+    toml_dict::CP.AbstractTOMLDict;
+    c::FT = toml_dict["moisture_stress_c"],
+    porosity_residual = true,
+    soil_params
+) where {FT <: AbstractFloat}
+
+Helper function to create `PiecewiseMoistureStressModel` by calculating field capacity (θ_high)
+and wilting point (θ_low) from the soil parameters.
+
+The soil parameters are expected to include ν and θ_r (if porosity_residual
+== true) AND  the van Genuchten parameters α and n (if porosity_residual == false). These
+may be fields or floats, but must be consistently one or the other.
+
+If porosity_residual == true, we use the porosity as θ_high, and the residual as θ_low; if false,
+we compute the air entry θ (θ_high) and use the intersection of the tangent of the van Genuchten curve at
+the inflection point with θ = 0 as θ_low.
```

Comment:
I'm not sure I understand this - if porosity_residual is true, alpha and n aren't used here so they don't strictly need to be provided right?

Could you explain more in the docstring about what porosity_residual corresponds to physically? It seems like it indicates whether to use the combination of soil porosity and the residual water content for the moisture stress thresholds, and to otherwise use "air_entry" and "esp_wp", but some more detail would be helpful. Also can you add more description of what the functions `compute_air_entry` and `compute_esp_wp` are calculating?